### ISSUE TITLE
Failed to collect receipts in exchange for query fee voucher: invalid BigNumber value

### ISSUE THREAD

**Issue by colindickson** [graphprotocol/indexer#675]
After I close an allocation or reallocate, I see the following error in my indexer logs:

```
{
  "level": 50,
  "time": 1688402149262,
  "pid": 1,
  "hostname": "indexer-agent-0",
  "name": "IndexerAgent",
  "component": "AllocationReceiptCollector",
  "allocation": "0x1F9A8C681D25E3Ec2a5934a10E0DA86815493F0f",
  "function": "obtainReceiptsVoucher()",
  "err": {
    "type": "IndexerError",
    "message": "Failed to collect receipts in exchange for query fee voucher",
    "stack": "IndexerError: Failed to collect receipts in exchange for query fee voucher\n    at indexerError (/opt/indexer/packages/indexer-common/dist/errors.js:171:12)\n    at AllocationReceiptCollector.<anonymous> (/opt/indexer/packages/indexer-common/dist/allocations/query-fees.js:322:147)\n    at Generator.throw (<anonymous>)\n    at rejected (/opt/indexer/packages/indexer-common/dist/allocations/query-fees.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
    "code": "IE054",
    "explanation": "https://github.com/graphprotocol/indexer/blob/main/docs/errors.md#ie054",
    "cause": {
      "type": "Error",
      "message": "invalid BigNumber value (argument=\"value\", value=undefined, code=INVALID_ARGUMENT, version=bignumber/5.7.0)",
      "stack": "Error: invalid BigNumber value (argument=\"value\", value=undefined, code=INVALID_ARGUMENT, version=bignumber/5.7.0)\n    at Logger.makeError (/opt/indexer/node_modules/@ethersproject/logger/lib/index.js:238:21)\n    at Logger.throwError (/opt/indexer/node_modules/@ethersproject/logger/lib/index.js:247:20)\n    at Logger.throwArgumentError (/opt/indexer/node_modules/@ethersproject/logger/lib/index.js:250:21)\n    at Function.BigNumber.from (/opt/indexer/node_modules/@ethersproject/bignumber/lib/bignumber.js:239:23)\n    at toBN (/opt/indexer/node_modules/@ethersproject/bignumber/lib/bignumber.js:292:25)\n    at BigNumber.add (/opt/indexer/node_modules/@ethersproject/bignumber/lib/bignumber.js:57:43)\n    at AllocationReceiptCollector.<anonymous> (/opt/indexer/packages/indexer-common/dist/allocations/query-fees.js:305:26)\n    at Generator.next (<anonymous>)\n    at fulfilled (/opt/indexer/packages/indexer-common/dist/allocations/query-fees.js:5:58)\n    at runMicrotasks (<anonymous>)",
      "reason": "invalid BigNumber value",
      "code": "INVALID_ARGUMENT",
      "argument": "value"
    }
  },
  "msg": "Failed to collect receipts in exchange for an on-chain query fee voucher"
}
```

version:

```
root@indexer-agent-0:/opt/indexer/packages/indexer-agent# cat package.json
{
  "name": "@graphprotocol/indexer-agent",
  "version": "0.20.17",
...
}
```

**Comment by madumas** [comment]
Encounted the same issue with 0.20.17. Downgrading to 0.20.16 allowed my indexer to redeem these vouchers.

**Comment by tilacog** [comment]
We included a patch for this in #668
